dot/node_integration_test.go

```diff
@@ -0,0 +1,329 @@
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	expected = NodeInitialized(cfg.Global.BasePath)
+	require.Equal(t, expected, true)
+}
+
+func TestNewNode(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := NewTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	err = keystore.LoadKeystore("alice", ks.Gran)
+	require.NoError(t, err)
+	err = keystore.LoadKeystore("alice", ks.Babe)
+	require.NoError(t, err)
+
+	cfg.Core.Roles = types.FullNodeRole
+
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	bp := node.Services.Get(&babe.Service{})
+	require.NotNil(t, bp)
+	fg := node.Services.Get(&grandpa.Service{})
+	require.NotNil(t, fg)
+}
+
+func TestNewNode_Authority(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := NewTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	err = keystore.LoadKeystore("alice", ks.Gran)
+	require.NoError(t, err)
+	require.Equal(t, 1, ks.Gran.Size())
+	err = keystore.LoadKeystore("alice", ks.Babe)
+	require.NoError(t, err)
+	require.Equal(t, 1, ks.Babe.Size())
+
+	cfg.Core.Roles = types.AuthorityRole
+
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	bp := node.Services.Get(&babe.Service{})
+	require.NotNil(t, bp)
+	fg := node.Services.Get(&grandpa.Service{})
+	require.NotNil(t, fg)
+}
+
+func TestStartNode(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := NewTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+	cfg.Core.GrandpaAuthority = false
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	err = keystore.LoadKeystore("alice", ks.Gran)
+	require.NoError(t, err)
+	err = keystore.LoadKeystore("alice", ks.Babe)
+	require.NoError(t, err)
+
+	cfg.Core.Roles = types.FullNodeRole
+
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	err = node.Start()
+	require.NoError(t, err)
+	<-node.started
+	node.Stop()
+}
+
+func TestInitNode_LoadGenesisData(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genPath := NewTestGenesisAndRuntime(t)
+
+	cfg.Init.Genesis = genPath
+	cfg.Core.GrandpaAuthority = false
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	config := state.Config{
+		Path: cfg.Global.BasePath,
+	}
+	stateSrvc := state.NewService(config)
+
+	gen, err := genesis.NewGenesisFromJSONRaw(genPath)
+	require.NoError(t, err)
+
+	genTrie, err := genesis.NewTrieFromGenesis(gen)
+	require.NoError(t, err)
+
+	genesisHeader, err := types.NewHeader(common.NewHash([]byte{0}), genTrie.MustHash(), trie.EmptyHash,
+		big.NewInt(0), types.NewDigest())
+	require.NoError(t, err)
+
+	err = stateSrvc.Initialise(gen, genesisHeader, genTrie)
+	require.NoError(t, err)
+
+	err = stateSrvc.Start()
+	require.NoError(t, err)
+
+	defer func() {
```

Comment:
Use `t.Cleanup` instead since this might not be run if `t.FailNow()` is called (by `require.*` functions)